src/arrayMethodLastIndexOf.js

```diff
@@ -5,8 +5,15 @@
  *
  */
 function applyCustomLastIndexOf() {
-  [].__proto__.lastIndexOf2 = function() {
-    // write code here
+  [].__proto__.lastIndexOf2 = function(value, fromIndex = this.length) {
+    if (value === undefined || this.length === 0) return -1;
+    if (fromIndex < 0) fromIndex = this.length + fromIndex;
```

Comment:
do not mutate function arguments.

If you need complex logic to declare iterator do it before for will be ok

Reply:
For my mind will be better to use const and ternary, but it is ok too if you prefer this way